### PR TITLE
qt 6.2.0

### PR DESCRIPTION
**Pull request by paperchalice**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Oops; meant to run that for `md4c` to see why it's not bottled.

**Comment by paperchalice** [comment]
I have reported it to upstream, a workaround is build  qt `gn` first.

File: Formula/qt.rb [hunk 1]
```diff
@@ -91,6 +88,35 @@ def install
...
+      # FIXME: GN requires clang in clangBasePath/bin
+      inreplace "qtwebengine/cmake/Functions.cmake", '"${clangBasePath}"', '"/usr"'
+      # Side effect by first `inreplace`
+      inreplace "qtwebengine/src/3rdparty/chromium/build/toolchain/mac/linker_driver.py",
+        "[dsymutil_path]", "['/usr/bin/dsymutil']"
+      inreplace "qtwebengine/src/3rdparty/chromium/build/toolchain/mac/BUILD.gn",
+        'rebase_path("$clang_base_path/bin/", root_build_dir)', '""'
+      arch = "x86_64"
+      arch = "arm64" if Hardware::CPU.arm?
+      gen = "#{buildpath}/qtwebengine/src/core/Release/#{arch}"
```
Comment (carlocab) [inline Formula/qt.rb:102]:
```suggestion
      gen = buildpath/"qtwebengine/src/core/Release"/Hardware::CPU.arch
```

File: Formula/qt.rb [hunk 2]
```diff
@@ -91,6 +88,35 @@ def install
...
+      inreplace "qtwebengine/src/core/configure/BUILD.root.gn.in",
+        "@GN_ARGS_INCLUDES@,",
+        "@GN_ARGS_INCLUDES@, \"-I#{gen}/gen/ABS_PATH#{gen}\","
```
Comment (carlocab) [inline Formula/qt.rb:105]:
```suggestion
      inreplace "qtwebengine/src/core/configure/BUILD.root.gn.in",
                "@GN_ARGS_INCLUDES@,",
                "@GN_ARGS_INCLUDES@, \"-I#{gen}/gen/ABS_PATH#{gen}\","
```

File: Formula/qt.rb [hunk 3]
```diff
@@ -91,6 +88,35 @@ def install
     # It is not friendly to Homebrew or macOS
     # because on macOS `/tmp` -> `/private/tmp`
     inreplace "qtbase/CMakeLists.txt", "FATAL_ERROR", ""
+    # WORKAROUND: build gn first
+    if OS.mac?
```
Comment (carlocab) [inline Formula/qt.rb:92]: Why do we need the third and fourth `inreplace`? Is there an upstream bug report?

Comment (paperchalice) [reply Formula/qt.rb:92]: In fact `qt` doesn't support build path containing symlink https://github.com/qt/qtbase/blob/a8823db57839f0a690f1cd488e2a268871aeca68/CMakeLists.txt#L21. It causes `gn` create some headers in an unexpected folder.

**Comment by maxcharlier** [comment]
as the ticket is closed, qt 6.2 will no be added to brew ?

**Comment by paperchalice** [comment]
A solution I found is setting `HOMEBREW_TEMP` to some path which doesn't contain the symlink, but the issue about `clang_base_path` is inevitable because the Homebrew shims directory layout is not what `GN` expect.

**Pull request closed without merging**